src/script.js

```diff
@@ -1,3 +1,47 @@
 "use strict";
 
 // TODO - write your code here.
+
+const randomDamage = () => Math.floor(Math.random() * 10) + 1;
+
+const chooseOption = (opt1, opt2) => {
+    const randNum = Math.round(Math.random());
+    return randNum === 0 ? opt1 : opt2;
+}
+
+const attackPlayer = (health) => health - randomDamage();
+
+const logHealth = (player, health) => console.log(`${player} health: ${health}`);
+
+const logDeath = (winner, loser) => console.log(`${winner} defeated ${loser}`);
+
+const isDead = (health) => health <= 0;
+
+const fight = (
+    player1, 
+    player2, 
+    player1Health, 
+    player2Health
+) => {
+	while (true) {
+		const attacker = chooseOption(player1, player2);
+
+    	if (attacker === player1) {
+			player2Health = attackPlayer(player2Health);
+        	logHealth(player2, player2Health);
+        	if (isDead(player2Health)) {
+        		logDeath(player1, player2);
+        		break;
+        	}
+    	} else {
+    		player1Health = attackPlayer(player1Health);
+        	logHealth(player1, player1Health);
+        	if (isDead(player1Health)) {
+            	logDeath(player2, player1);
+            	break;
+	        }
+    	}
+    }
+}
```

Comment:
styling: take care to apply consistent tab/spacing for your code. there's a mix of tabs (the specific whitespace character) and spaces. This is something that  can be configured in VS Code. David made the suggestion to rely on tab characters, so you can follow that. But generally, whichever approach you use, I encourage you to take a moment before you're "done done" to visually inspect your code to ensure consistent spacing. Either 2 or 4 spaces, or consistent tabs only.